packages/pages/src/scaffold/template/sampleTemplates.ts

```diff
@@ -0,0 +1,128 @@
+// TODO: Use new resolveData (or similar function) in transformProps
+export const visualEditorTemplateCode = (
+  templateName: string,
+  fileName: string,
+  entityScope: string,
+  filter: string[]
+): string => {
+  const formattedTemplateName =
+    fileName.charAt(0).toUpperCase() + fileName.slice(1);
+  const filterCode = `${entityScope}: ${JSON.stringify(filter)},`;
+  const config = `${fileName}Config`;
+
+  return `import {
+  Template,
+  GetPath,
+  TemplateConfig,
+  TemplateProps,
+  TemplateRenderProps,
+  GetHeadConfig,
+  HeadConfig,
+} from "@yext/pages";
+import { Config, Render } from "@measured/puck";
+import { ${config} } from "../ve.config";
+import { DocumentProvider } from "../hooks/useDocument";
+import { getTemplatePuckData } from "../utils/puckDataHelper";
+
+export const config: TemplateConfig = {
+  name: "${templateName}",
+  stream: {
+    $id: "${templateName}-stream",
+    filter: {
+      ${filterCode}
+    },
+    fields: [
+      "id",
+      "name",
+      "slug",
+      "c_visualConfigurations",
+      "c_pages_layouts.c_visualConfiguration",
+    ],
+    localization: {
+      locales: ["en"],
+    },
+  },
+  additionalProperties: {
+    isVETemplate: true,
+    isDraft: true,
+  }
+};
+
+export const transformProps = async (data) => {
+  const { document } = data;
+  const entityConfigurations = document.c_visualConfigurations ?? [];
+  const entityLayoutConfigurations = document.c_pages_layouts ?? [];
+  const siteLayoutConfigurations = document._site?.c_visualLayouts;
+  try {
```

Comment:
All of this will end up being a one line function call to a function from the ve lib

Reply:
using it from this [PR](https://github.com/yext/visual-editor/pull/6/), will update if changed